packages/ag-charts-community/src/util/function.ts

```diff
@@ -72,7 +74,7 @@ export function debounce<T extends (...args: Parameters<T>) => void>(
     });
 }
 
-export function throttle<T extends (...args: Parameters<T>) => void>(
+export function throttle<T extends (...args: Parameters<T>) => void | Promise<void>>(
```

Comment:
`throttle`callback shouldn't return a promise, it's a bit of an anti-pattern.
I think it's better if we take care of it on the `dataService` side instead of the generic method.

Reply:
Fair, I've pushed the error handling back out to the callback.